ical/parsing/parser.py

```diff
@@ -25,96 +25,133 @@
 """
 
 import logging
-import threading
 from functools import cache
-from typing import cast
+import re
 from typing import Iterable
-
-from pyparsing import (
-    Combine,
-    Group,
-    Or,
-    ParserElement,
-    ParseResults,
-    QuotedString,
-    Word,
-    ZeroOrMore,
-    alphanums,
-    alphas,
-    nums,
-    ParseException,
-)
-
 from ical.exceptions import CalendarParseError
-
-from .const import (
-    PARSE_NAME,
-    PARSE_PARAM_NAME,
-    PARSE_PARAM_VALUE,
-    PARSE_PARAMS,
-    PARSE_VALUE,
-)
-from .unicode import SAFE_CHAR, VALUE_CHAR
-
+from ical.parsing.property import ParsedProperty, ParsedPropertyParameter
 
 _LOGGER = logging.getLogger(__name__)
 
+_RE_CONTROL_CHARS = re.compile("[\x00-\x08\x0A-\x1F\x7F]")
+_RE_NAME = re.compile("[A-Z0-9-]+")
 
 @cache
-def _create_parser() -> ParserElement:
-    """Create rfc5545 parser."""
-    iana_token = Word(alphanums + "-")
-    vendor_id = Word(alphanums)
-    x_name = Combine("X-" + ZeroOrMore(vendor_id + "-") + alphas + nums + "-")
-    name = Or([iana_token, x_name])
-
-    param_name = Or([iana_token, x_name])
-    # rfc5545 Q-SAFE-CHAR is any character except CONTROL and DQUOTE which is
-    # close enough to the pyparsing provided parser element
-    param_text = Word(SAFE_CHAR) | ""
-    quoted_string = QuotedString('"')
-
-    param_value = Or([param_text, quoted_string])
-    # There are multiple levels of property parameter grouping since there can
-    # either be repreated property parameters with the same name or property
-    # s parameters with repeated values. A two level structure is used to grab
-    # both, which is then flattened when consuming the result.
-    param = Group(
-        param_name.set_results_name(PARSE_PARAM_NAME)
-        + "="
-        + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
-        + ZeroOrMore(
-            "," + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
+def parse_line(line: str) -> ParsedProperty:
+    """Parse a single line."""
+    
+    result = ParsedProperty(name="", value="", params=None)
```

Comment:
Let's keep the ParsedProperty immutable and construct it at the end as the return value, similar to what I was saying about the dict return result.